src/security/ServerOptions.cc

```diff
@@ -244,9 +244,20 @@ Security::ServerOptions::createStaticServerContext(AnyP::PortCfg &port)
 
 #elif USE_GNUTLS
         for (auto &keys : certs) {
-            gnutls_x509_crt_t crt = keys.cert.get();
             gnutls_x509_privkey_t xkey = keys.pkey.get();
-            const auto x = gnutls_certificate_set_x509_key(t.get(), &crt, 1, xkey);
+
+            const auto certCount = 1 + keys.chain.size();
+            gnutls_x509_crt_t *crt = new gnutls_x509_crt_t[certCount];
```

Comment:
Please use an exception-safe std::array here if possible. Otherwise:

```suggestion
            auto crt = new gnutls_x509_crt_t[certCount];
```

Reply:
std::array is not possible because "error: the value of certCount is not usable in a constant expression"